### PR TITLE
Dependecies in setuptools and cleanup for romanian flavor

### PR DESCRIPTION
**Pull request by mnemonicflow**
The changes include the following:
1. Added required and optional dependencies inside setup.py 
So that pip will install django if it is missing and when building the documentation to install sphinx 
Let me know if this is not needed and I will remove the commit from the PR.
2. Improved the docstrings for the romanian flavour (removed broken links, added missing docstrings, added clarifications)
3. Reformated the code in the romanian flavor forms.py file for increased readability

### CODE REVIEW

**Comment by benkonrath** [comment]
@mnemonicflow Can you split out the changes to setup.py into a separate PR. I don't do the releases so I don't feel comfortable making changes there. I'll review your cleanups to the ro module once this is done. Thanks.

**Comment by mnemonicflow** [comment]
@benkonrath I've updated the pull request and removed the commit related to setup.py

File: localflavor/ro/forms.py [hunk 1]
```diff
@@ -35,32 +29,40 @@ def __init__(self, max_length=10, min_length=2, *args, **kwargs):
...
             raise ValidationError(self.error_messages['invalid'])
+
         return value[::-1]
 
 
 class ROCNPField(RegexField):
     """
     A Romanian personal identity code (CNP) field
-
-    For CNP validation algorithm see http://www.validari.ro/cnp.html
```
Comment (benkonrath) [inline localflavor/ro/forms.py]: Is there another website that has this information (preferably in English)?

File: localflavor/ro/forms.py [hunk 2]
```diff
@@ -73,26 +75,36 @@ def __init__(self, max_length=13, min_length=13, *args, **kwargs):
...
+
+        Args:
+            value: the CNP code
         """
         value = super(ROCNPField, self).clean(value)
         if value in EMPTY_VALUES:
             return ''
-        # check birthdate digits
+
+        # check birthday digits
```
Comment (benkonrath) [inline localflavor/ro/forms.py]: `birthdate` is correct if the check includes the year. 

https://english.stackexchange.com/a/210491

File: localflavor/ro/forms.py [hunk 3]
```diff
@@ -3,26 +3,20 @@
...
 import re
-
 from django.core.validators import EMPTY_VALUES
 from django.forms import ValidationError, Field, RegexField, Select
 from django.utils.translation import ugettext_lazy as _
-
 from ..generic.forms import IBANFormField
 from .ro_counties import COUNTIES_CHOICES
 
-
```
Comment (benkonrath) [inline localflavor/ro/forms.py]: Please keep the spaces in the imports to match the rest of localflavor. A proper solution to getting consistent imports would be to setup something with isort for the whole project but there are other things that I think are more important right now. Thanks.

**Comment by benkonrath** [comment]
@mnemonicflow Thanks for your work here. Just those few comments. The rest looks good to me.

**Comment by mnemonicflow** [comment]
@benkonrath Can you please review the code now?

**Comment by benkonrath** [comment]
Thanks.

**Pull request merged**